clang-tools-extra/clangd/test/module_dependencies.test

```diff
@@ -0,0 +1,96 @@
+# A smoke test to check that a simple dependency chain for modules can work.
+#
+# FIXME: This fails on the Windows ARM64 build server. Not entirely sure why as it has been tested on
+#        an ARM64 Windows VM and appears to work there.
+# UNSUPPORTED: host=aarch64-pc-windows-msvc
+#
+# RUN: rm -fr %t
+# RUN: mkdir -p %t
+# RUN: split-file %s %t
+#
+# RUN: sed -e "s|DIR|%/t|g" %t/compile_commands.json.tmpl > %t/compile_commands.json.tmp
+# RUN: sed -e "s|CLANG_CC|%clang|g" %t/compile_commands.json.tmp > %t/compile_commands.json
+# RUN: sed -e "s|DIR|%/t|g" %t/definition.jsonrpc.tmpl > %t/definition.jsonrpc.tmp
```

Comment:
Using `sed` for paths like this isn't portable, and I'm seeing failures on some internal windows bots because of it. The `UNSUPPORTED` here doesn't seem sufficient.

Notably, since on windows paths will include backslashes, a number of paths will be interpreted as having escape sequences, so we end up with either an error from the sed command saying we can't use that escape sequence there, or some kind of garbage in the replacement text.

Reply:
@fleeting-xx I suspect the right medium to long term solution here would be to convert these tests to unit tests, as @ChuanqiXu9 suggested. This would avoid having to mess around with the config file using `sed` since all of that set up could be done programatically.

Thanks @zmodem for the short term fix!